controllers/btpoperator_controller.go

```diff
@@ -1047,6 +1047,11 @@ func (r *BtpOperatorReconciler) softDelete(ctx context.Context, gvk schema.Group
 
 	isBinding := gvk.Kind == btpOperatorServiceBinding
 	for _, item := range list.Items {
+		if item.GetDeletionTimestamp().IsZero() {
+			if err := r.Delete(ctx, &item); err != nil {
+				return err
+			}
+		}
```

Comment:
previously it was only `hardDelete()` that can call deletion on btp-operator resources. However, there is a timeout on `hardDelete()` that can expire before resource deletion is called or can error and never call the deletion, resulting in resources having no `finalizer` but also no `deletionTimestamp` when going through `softDelete()`.

This redundant resource deletion checks if there is no `deletionTimestamp` and ensures it's called at least once for each resource. This improves the robustness of `softDelete` and allows shorter timeout for tests.